diff --git a/node_modules/@react-native/codegen/lib/parsers/flow/components/componentsUtils.js b/node_modules/@react-native/codegen/lib/parsers/flow/components/componentsUtils.js
index b693ba8..f5830d9 100644
--- a/node_modules/@react-native/codegen/lib/parsers/flow/components/componentsUtils.js
+++ b/node_modules/@react-native/codegen/lib/parsers/flow/components/componentsUtils.js
@@ -41,7 +41,7 @@ function getTypeAnnotationForArray(
   if (extractedTypeAnnotation.type === 'GenericTypeAnnotation') {
     // Resolve the type alias if it's not defined inline
     const objectType = getValueFromTypes(extractedTypeAnnotation, types);
-    if (objectType.id.name === '$ReadOnly') {
+    if (objectType.id.name === '$ReadOnly' || objectType.id.name === 'Readonly') {
       return {
         type: 'ObjectTypeAnnotation',
         properties: flattenProperties(
@@ -53,7 +53,7 @@ function getTypeAnnotationForArray(
           .filter(Boolean),
       };
     }
-    if (objectType.id.name === '$ReadOnlyArray') {
+    if (objectType.id.name === '$ReadOnlyArray' || objectType.id.name === 'ReadonlyArray') {
       // We need to go yet another level deeper to resolve
       // types that may be defined in a type alias
       const nestedObjectType = getValueFromTypes(
@@ -206,7 +206,8 @@ function getTypeAnnotation(
   const typeAnnotation = getValueFromTypes(annotation, types);
   if (
     typeAnnotation.type === 'GenericTypeAnnotation' &&
-    parser.getTypeAnnotationName(typeAnnotation) === '$ReadOnlyArray'
+    (parser.getTypeAnnotationName(typeAnnotation) === '$ReadOnlyArray' ||
+      parser.getTypeAnnotationName(typeAnnotation) === 'ReadonlyArray')
   ) {
     return {
       type: 'ArrayTypeAnnotation',
@@ -222,7 +223,8 @@ function getTypeAnnotation(
   }
   if (
     typeAnnotation.type === 'GenericTypeAnnotation' &&
-    parser.getTypeAnnotationName(typeAnnotation) === '$ReadOnly'
+    (parser.getTypeAnnotationName(typeAnnotation) === '$ReadOnly' ||
+      parser.getTypeAnnotationName(typeAnnotation) === 'Readonly')
   ) {
     return {
       type: 'ObjectTypeAnnotation',
diff --git a/node_modules/@react-native/codegen/lib/parsers/flow/components/events.js b/node_modules/@react-native/codegen/lib/parsers/flow/components/events.js
index c0b8291..894c113 100644
--- a/node_modules/@react-native/codegen/lib/parsers/flow/components/events.js
+++ b/node_modules/@react-native/codegen/lib/parsers/flow/components/events.js
@@ -49,6 +49,7 @@ function getPropertyType(
     case 'Float':
       return emitFloatProp(name, optional);
     case '$ReadOnly':
+    case 'Readonly':
       return getPropertyType(
         name,
         optional,
@@ -69,6 +70,7 @@ function getPropertyType(
       return emitMixedProp(name, optional);
     case 'ArrayTypeAnnotation':
     case '$ReadOnlyArray':
+    case 'ReadonlyArray':
       return {
         name,
         optional,
@@ -132,6 +134,7 @@ function extractArrayElementType(typeAnnotation, name, parser) {
         ),
       };
     case '$ReadOnlyArray':
+    case 'ReadonlyArray':
       const genericParams = typeAnnotation.typeParameters.params;
       if (genericParams.length !== 1) {
         throw new Error(
@@ -169,7 +172,7 @@ function findEventArgumentsAndType(
 ) {
   throwIfEventHasNoName(typeAnnotation, parser);
   const name = parser.getTypeAnnotationName(typeAnnotation);
-  if (name === '$ReadOnly') {
+  if (name === '$ReadOnly' || name === 'Readonly') {
     return {
       argumentProps: typeAnnotation.typeParameters.params[0].properties,
       paperTopLevelNameDeprecated: paperName,
diff --git a/node_modules/@react-native/codegen/lib/parsers/typescript/components/componentsUtils.js b/node_modules/@react-native/codegen/lib/parsers/typescript/components/componentsUtils.js
index 3ba47fe..2e2c97c 100644
--- a/node_modules/@react-native/codegen/lib/parsers/typescript/components/componentsUtils.js
+++ b/node_modules/@react-native/codegen/lib/parsers/typescript/components/componentsUtils.js
@@ -294,16 +294,14 @@ function getTypeAnnotationForArray(
     return arrayType;
   }
   const type =
-    extractedTypeAnnotation.elementType === 'TSTypeReference'
-      ? extractedTypeAnnotation.elementType.typeName.name
+    (extractedTypeAnnotation.elementType &&
+      extractedTypeAnnotation.elementType.type === 'TSTypeReference')
+      ? parser.getTypeAnnotationName(extractedTypeAnnotation.elementType)
       : ((_extractedTypeAnnotat = extractedTypeAnnotation.elementType) ===
           null || _extractedTypeAnnotat === void 0
           ? void 0
           : _extractedTypeAnnotat.type) ||
-        ((_extractedTypeAnnotat2 = extractedTypeAnnotation.typeName) === null ||
-        _extractedTypeAnnotat2 === void 0
-          ? void 0
-          : _extractedTypeAnnotat2.name) ||
+        parser.getTypeAnnotationName(extractedTypeAnnotation) ||
         extractedTypeAnnotation.type;
   const common = getCommonTypeAnnotation(
     name,
diff --git a/node_modules/@react-native/codegen/lib/parsers/typescript/components/extends.js b/node_modules/@react-native/codegen/lib/parsers/typescript/components/extends.js
index 4c1f723..dbc9bfa 100644
--- a/node_modules/@react-native/codegen/lib/parsers/typescript/components/extends.js
+++ b/node_modules/@react-native/codegen/lib/parsers/typescript/components/extends.js
@@ -16,8 +16,12 @@ function isEvent(typeAnnotation) {
   if (typeAnnotation.type !== 'TSTypeReference') {
     return false;
   }
+  const name =
+    typeAnnotation.typeName.type === 'TSQualifiedName'
+      ? typeAnnotation.typeName.right.name
+      : typeAnnotation.typeName.name;
   const eventNames = new Set(['BubblingEventHandler', 'DirectEventHandler']);
-  return eventNames.has(typeAnnotation.typeName.name);
+  return eventNames.has(name);
 }
 
 // $FlowFixMe[unclear-type] TODO(T108222691): Use flow-types for @babel/parser
diff --git a/node_modules/@react-native/codegen/lib/parsers/typescript/parseTopLevelType.js b/node_modules/@react-native/codegen/lib/parsers/typescript/parseTopLevelType.js
index 0ef998d..84fd4f3 100644
--- a/node_modules/@react-native/codegen/lib/parsers/typescript/parseTopLevelType.js
+++ b/node_modules/@react-native/codegen/lib/parsers/typescript/parseTopLevelType.js
@@ -88,10 +88,14 @@ function handleUnionAndParen(type, result, knownTypes) {
       }
       break;
     }
-    case 'TSTypeReference':
-      if (type.typeName.name === 'Readonly') {
+    case 'TSTypeReference': {
+      const typeName =
+        type.typeName.type === 'TSQualifiedName'
+          ? type.typeName.right.name
+          : type.typeName.name;
+      if (typeName === 'Readonly') {
         handleUnionAndParen(type.typeParameters.params[0], result, knownTypes);
-      } else if (type.typeName.name === 'WithDefault') {
+      } else if (typeName === 'WithDefault') {
         if (result.optional) {
           throw new Error(
             'WithDefault<> is optional and does not need to be marked as optional. Please remove the union of undefined and/or null',
@@ -124,6 +128,7 @@ function handleUnionAndParen(type, result, knownTypes) {
         }
       }
       break;
+    }
     default:
       result.unions.push(type);
   }
@@ -165,14 +170,18 @@ function handleIntersectionAndParen(type, result, knownTypes) {
       }
       break;
     }
-    case 'TSTypeReference':
-      if (type.typeName.name === 'Readonly') {
+    case 'TSTypeReference': {
+      const typeName =
+        type.typeName.type === 'TSQualifiedName'
+          ? type.typeName.right.name
+          : type.typeName.name;
+      if (typeName === 'Readonly') {
         handleIntersectionAndParen(
           type.typeParameters.params[0],
           result,
           knownTypes,
         );
-      } else if (type.typeName.name === 'WithDefault') {
+      } else if (typeName === 'WithDefault') {
         throw new Error('WithDefault<> is now allowed in intersection types.');
       } else if (!knownTypes) {
         result.push(type);
@@ -188,6 +197,7 @@ function handleIntersectionAndParen(type, result, knownTypes) {
         }
       }
       break;
+    }
     default:
       result.push(type);
   }
diff --git a/node_modules/@react-native/codegen/lib/parsers/typescript/parser.js b/node_modules/@react-native/codegen/lib/parsers/typescript/parser.js
index da4afe3..1faf982 100644
--- a/node_modules/@react-native/codegen/lib/parsers/typescript/parser.js
+++ b/node_modules/@react-native/codegen/lib/parsers/typescript/parser.js
@@ -100,13 +100,23 @@ class TypeScriptParser {
     return 'TypeScript';
   }
   getTypeAnnotationName(typeAnnotation) {
-    var _typeAnnotation$typeN;
-    return typeAnnotation === null ||
+    var _typeAnnotation$typeN, _typeAnnotation$typeN2;
+    if (
+      (typeAnnotation === null ||
       typeAnnotation === void 0 ||
       (_typeAnnotation$typeN = typeAnnotation.typeName) === null ||
       _typeAnnotation$typeN === void 0
+        ? void 0
+        : _typeAnnotation$typeN.type) === 'TSQualifiedName'
+    ) {
+      return typeAnnotation.typeName.right.name;
+    }
+    return typeAnnotation === null ||
+      typeAnnotation === void 0 ||
+      (_typeAnnotation$typeN2 = typeAnnotation.typeName) === null ||
+      _typeAnnotation$typeN2 === void 0
       ? void 0
-      : _typeAnnotation$typeN.name;
+      : _typeAnnotation$typeN2.name;
   }
   checkIfInvalidModule(typeArguments) {
     return (
